src/midend/mir_emitter.cpp

```diff
@@ -581,15 +588,20 @@ namespace graphit {
                 //dense vertexset apply
                 auto vertexset_apply_expr = std::make_shared<mir::VertexSetApplyExpr>();
                 vertexset_apply_expr->target = target_expr;
-                vertexset_apply_expr->input_function_name = apply_expr->input_function->ident;
+                vertexset_apply_expr->input_function_name = apply_expr->input_function->name->ident;
+                vertexset_apply_expr->functorArgs = emitFunctorArgs(apply_expr->input_function->args);
+
                 if (apply_expr->change_tracking_field != nullptr)
                     vertexset_apply_expr->tracking_field = fir::to<fir::Identifier>(
                             apply_expr->change_tracking_field)->ident;
                 retExpr = vertexset_apply_expr;
             } else if (apply_expr->type == fir::ApplyExpr::Type::UPDATE_PRIORITY_EXTERN_APPLY) {
                 auto apply_update_priority_expr = std::make_shared<mir::UpdatePriorityExternVertexSetApplyExpr>();
                 apply_update_priority_expr->target = target_expr;
-                apply_update_priority_expr->input_function_name = apply_expr->input_function->ident;
+                apply_update_priority_expr->input_function_name = apply_expr->input_function->name->ident;
+                apply_update_priority_expr->functorArgs = emitFunctorArgs(apply_expr->input_function->args);
```

Comment:
We need a test case for 
edges.applyUpdatePriority(functor[functor_arg])
See https://github.com/GraphIt-DSL/graphit/blob/master/apps/sssp_delta_stepping.gt

Reply:
sounds good! will do